server/app/src/main/java/com/google/heartrate/wearos/app/gatt/heartrate/characteristics/HeartRateMeasurementCharacteristic.java

```diff
@@ -0,0 +1,269 @@
+ * <p>
+ * Fields included in the characteristic:
+ * <table>
+ *  <thead>
+ *      <tr><th>Field</th><th>Requirement</th><th>Format</th></tr>
+ *  <thead>
+ *  <tbody>
+ *      <tr><td> Flags </td><td> Mandatory </td><td> UInt8 </td></tr>
+ *      <tr><td> Heart Rate Measurement </td><td> Mandatory </td><td> UInt8/UInt16 </td></tr>
+ *      <tr><td> Energy Expended </td><td> Optional, Mandatory if Energy Expended flag </td><td> UInt16 </td></tr>
+ *      <tr><td> RR-Interval </td><td> Optional, Not supported </td><td> - </td></tr>
+ *      <tr><td> Transmission Interval </td><td> Optional, Not supported </td><td> - </td></tr>
+ *  </tbody>
+ * </table>
+ * <p>
+ * See <a href="https://www.bluetooth.com/wp-content/uploads/Sitecore-Media-Library/Gatt/Xml/Characteristics/org.bluetooth.characteristic.heart_rate_measurement.xml".>
+ * Heart Rate Measurement Characteristic</a>.
+ */
+public class HeartRateMeasurementCharacteristic extends GattCharacteristic {
+    private static final String TAG = HeartRateMeasurementCharacteristic.class.getCanonicalName();
+
+    /** Heart Rate Measurement characteristic UUID. */
+    private static final UUID HEART_RATE_MEASUREMENT_UUID = UUID
+            .fromString("00002a37-0000-1000-8000-00805f9b34fb");
+
+    /** Indicates that the data format of the Heart Rate Measurement value field is in a format of UINT8.  */
+    private static final byte HEART_RATE_MEASUREMENT_UINT8_FLAG = 0b00000000;
+
+    /** Indicates that the data format of the Heart Rate Measurement value field is in a format of UINT16. */
+    private static final byte HEART_RATE_MEASUREMENT_UINT16_FLAG = 0b00000001;
+
+    /** Indicates that the Energy Expended field is present in the Heart Rate Measurement characteristic. */
+    private static final byte EXPENDED_ENERGY_FLAG = 0b00001000;
+
+
+    /** Offset at which the Heart Rate Measurement characteristic flags are stored. */
+    private static final int FLAGS_OFFSET = 0;
+
+    /** Offset at which Heart Rate Measurement value is stored. */
+    private static final int HEART_RATE_MEASUREMENT_OFFSET = 1;
+
+    /** Offset at which Expended Energy value is Heart Rate Measurement value field is in a format of UINT8. */
+    private static final int EXPENDED_ENERGY_BASE_OFFSET = 2;
+
+    /** Offset at which Expended Energy value is Heart Rate Measurement value field is in a format of UINT16. */
+    private static final int EXPENDED_ENERGY_SHIFTED_OFFSET = 3;
+
+    /**
+     * Create {@link GattCharacteristic} for Heart Rate Measurement characteristic.
+     * <br>Configure {@link GattCharacteristic} with Heart Rate Measurement characteristic UUID,
+     * property for notification, no read/write permissions and Client Characteristic Configuration Descriptor.
+     */
+    public HeartRateMeasurementCharacteristic() {
+        super(HEART_RATE_MEASUREMENT_UUID,
+                BluetoothGattCharacteristic.PROPERTY_NOTIFY,
+                BluetoothGattCharacteristic.PROPERTY_NOTIFY,
```

Comment:
I think here should be the permissions on the characteristic, e.g. PERMISSION_READ etc.

Reply:
Fixed in [commit](https://github.com/googleinterns/heartrate-bt-wear/pull/1/commits/5a31602eddf7e1d410e7bf33fc2093a9c51e5382). Set no read/write permissions for heart rate measurement characteristic.